kernel/src/services.rs

```diff
@@ -1141,6 +1161,27 @@ impl SystemServices {
             };
             // log_process_update(file!(), line!(), new, old_state);
             new.activate()?;
+            match lazy_arg {
+                PostActivateOp::RememberServerMessage(
+                    sidx,
+                    current_pid,
+                    current_thread,
+                    message,
+                    client_address,
+                ) => {
+                    let si = {
+                        let server =
+                            self.server_from_sidx_mut(sidx).expect("couldn't re-discover server index");
+                        server.queue_response(current_pid, current_thread, message, client_address)?
+                    };
+                    sender_idx = Some(si);
+                }
+                PostActivateOp::SetThreadResult(result) => {
+                    let mut arch_process = ArchProcess::current();
+                    arch_process.set_thread_result(new_tid, result);
+                }
+                _ => (),
```

Comment:
It might be good to make this an exhaustive match, in case another op is added later.
```suggestion
                None => (),
```